src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/PackageSourceMappingUtility.cs

```diff
@@ -91,5 +91,52 @@ internal static class PackageSourceMappingUtility
 
             mappingProvider.SavePackageSourceMappings(newAndExistingPackageSourceMappingItems);
         }
+
+        internal static string? GetNewSourceMappingSourceName(PackageSourceMapping packageSourceMapping, PackageSourceMoniker activePackageSourceMoniker)
+        {
+            string? sourceMappingSourceName = packageSourceMapping.IsEnabled
+                && !activePackageSourceMoniker.IsAggregateSource
+                ? activePackageSourceMoniker.PackageSourceNames.First() : null;
+
+            return sourceMappingSourceName;
+        }
+
+        internal static void GetNewSourceMappingsFromAddedPackages(ref Dictionary<string, SortedSet<string>>? newSourceMappings, string newMappingSourceName, List<AccessiblePackageIdentity> added, PackageSourceMapping packageSourceMapping)
```

Comment:
From an API design point of view, `Get` suggests a return value. I think `Add` better represents that the method is adding to a dictionary (well, unless `null` was passed in, which which case it creates a dictionary first)

```suggestion
        internal static void AddNewSourceMappingsFromAddedPackages(ref Dictionary<string, SortedSet<string>>? newSourceMappings, string newMappingSourceName, List<AccessiblePackageIdentity> added, PackageSourceMapping packageSourceMapping)
```

Reply:
Good catch. I missed that.

Reply:
Absolutely! As I reworked the implementation, the name got a little stale.